pkg/project/project.go

```diff
@@ -232,6 +240,10 @@ func (p *Project) CommitUUID() strfmt.UUID {
 	return strfmt.UUID(p.commitID)
 }
 
+func (p *Project) IsHeadless() bool {
+	return p.commitID != "" && (p.name == "" || p.owner == "")
```

Comment:
Let's instead just check if the url contains `/commit/` (ie. match the CommitURLRe). Technically this logic here is sound but I prefer we be a bit more explicit.